subsys/bluetooth/audio/shell/audio.h

```diff
@@ -289,111 +273,35 @@ static inline void copy_unicast_stream_preset(struct shell_stream *stream,
-		struct bt_audio_codec_data *data = &stream->codec_cfg.meta[i];
-		const uint8_t data_len = preset_data->data.data_len;
-
-		data->data.data = data->value;
-		data->data.data_len = data_len;
-		memcpy(data->value, preset_data->data.data, data_len);
-	}
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT > 0 &&                                      \
-	* CONFIG_BT_AUDIO_CODEC_MAX_DATA_LEN > 0                                                   \
-	*/
 }
 
 static inline void copy_broadcast_source_preset(struct broadcast_source *source,
 						const struct named_lc3_preset *named_preset)
 {
 	memcpy(&source->qos, &named_preset->preset.qos, sizeof(source->qos));
 	memcpy(&source->codec_cfg, &named_preset->preset.codec_cfg, sizeof(source->codec_cfg));
-
-#if CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT > 0 && CONFIG_BT_AUDIO_CODEC_MAX_DATA_LEN > 0
-	/* Need to update the `bt_data.data` pointer to the new value after copying the codec */
-	for (size_t i = 0U; i < ARRAY_SIZE(source->codec_cfg.data); i++) {
-		const struct bt_audio_codec_data *preset_data =
-			&named_preset->preset.codec_cfg.data[i];
-		struct bt_audio_codec_data *data = &source->codec_cfg.data[i];
-		const uint8_t data_len = preset_data->data.data_len;
-
-		data->data.data = data->value;
-		data->data.data_len = data_len;
-		memcpy(data->value, preset_data->data.data, data_len);
-	}
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT > 0 && CONFIG_BT_AUDIO_CODEC_MAX_DATA_LEN > \
-	* 0                                                                                        \
-	*/
-
-#if CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT > 0 && CONFIG_BT_AUDIO_CODEC_MAX_DATA_LEN > 0
-	for (size_t i = 0U; i < ARRAY_SIZE(source->codec_cfg.meta); i++) {
-		const struct bt_audio_codec_data *preset_data =
-			&named_preset->preset.codec_cfg.meta[i];
-		struct bt_audio_codec_data *data = &source->codec_cfg.meta[i];
-		const uint8_t data_len = preset_data->data.data_len;
-
-		data->data.data = data->value;
-		data->data.data_len = data_len;
-		memcpy(data->value, preset_data->data.data, data_len);
-	}
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT > 0 && CONFIG_BT_AUDIO_CODEC_MAX_DATA_LEN > \
-	* 0                                                                                        \
-	*/
-}
-
-static inline void codec_data_set_chan_alloc(struct bt_audio_codec_data *data,
-					     enum bt_audio_location loc)
-{
-	const uint32_t loc_32 = loc;
-
-	data->data.type = BT_AUDIO_CODEC_CONFIG_LC3_CHAN_ALLOC;
-	data->data.data_len = sizeof(loc_32);
-	sys_put_le32(loc_32, data->value);
 }
 
 static inline int codec_set_chan_alloc(struct bt_audio_codec_cfg *codec_cfg,
 				       enum bt_audio_location loc)
 {
-	for (size_t i = 0U; i < codec_cfg->data_count; i++) {
-		struct bt_audio_codec_data *data = &codec_cfg->data[i];
+	for (size_t i = 0U; i < codec_cfg->data_len;) {
+		const uint8_t len = codec_cfg->data[i++];
+		const uint8_t type = codec_cfg->data[i++];
```

Comment:
Is there a potential for i to overflow the data length (if faulty data is injected)?

Reply:
Yes, will need to add a check for `len` everywhere - Thanks